### PR TITLE
fix retiarii config experiment_working_directory

### PR DESCRIPTION
**Pull request by J-shang**

### CODE REVIEW

File: nni/retiarii/experiment/pytorch.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ class RetiariiExeConfig(ConfigBase):
     nni_manager_ip: Optional[str] = None
     debug: bool = False
     log_level: Optional[str] = None
-    experiment_working_directory: Optional[PathLike] = None
+    experiment_working_directory: Optional[PathLike] = '~/nni-experiments'
```
Comment (ultmaster) [inline nni/retiarii/experiment/pytorch.py:55]: I'm not sure. What will happen if it's none?

Comment (J-shang) [reply nni/retiarii/experiment/pytorch.py:55]: if we don't have a default value and the user don't set it, we will get
image

Comment (ultmaster) [reply nni/retiarii/experiment/pytorch.py:55]: I had realized the problem. But I'm not sure whether this is a correct fix.

Comment (J-shang) [reply nni/retiarii/experiment/pytorch.py]: Yes, is `PathLike`.

We have similar ExperimentConfig, FYI
https://github.com/microsoft/nni/blob/02eab99b9bb280e385a7af71b4c6c4a73bae1f04/nni/experiment/config/common.py#L69

Comment (ultmaster) [reply nni/retiarii/experiment/pytorch.py]: But I just tested. It only works when I changed `~/nni-experiments` to `/home/xxx/nni-...`.

Have you checked the two blue buttons to view logs?

Comment (J-shang) [reply nni/retiarii/experiment/pytorch.py]: Yes, I can view the log, and `expanduser` is here:

https://github.com/microsoft/nni/blob/02eab99b9bb280e385a7af71b4c6c4a73bae1f04/nni/experiment/config/base.py#L60

Comment (ultmaster) [reply nni/retiarii/experiment/pytorch.py]: Could it be an issue with `Optional`? I'll retest it when this PR gets merged.

**Review by ultmaster — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**